### PR TITLE
add pre-OOP version of world

### PR DESCRIPTION
**Pull request by Roophee**
### Tyni JS World pre-OOP

[Demo](https://roophee.github.io/js-tyni-world-noOOP/)
[Code base](https://github.com/Roophee/frontend-2021-homeworks/tree/a-tyni-js-world-2/submissions/Roophee/a-tyni-js-world-roophee)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/Roophee/a-tyni-js-world-roophee/index.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    friends: ['Sam', 'Hanna', 'Kitty', 'Milo']
+}
+
+let toStringTask = function (item) {
+    let string = '';
+    for (let i in item) {
+        string += item[i] + ';';
+    }
+    return string;
+}
```
Comment (OleksiyRudenko) [inline submissions/Roophee/a-tyni-js-world-roophee/index.js:62]: Please check yourself against requirements.

image

`for...in` is a variant of `Object.keys()`

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@Roophee nicely done.
However, there is an issue to fix yet.

File: submissions/Roophee/a-tyni-js-world-roophee/index.js (hunk 1 above)
Comment (Roophee) [reply submissions/Roophee/a-tyni-js-world-roophee/index.js]: Sorry for this and thanks for your patience. 

Fix this. Rewrite method that create string.

File: submissions/Roophee/a-tyni-js-world-roophee/index.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    gender: 'female',
+    legs: 2,
+    hands: 2,
+    tail: 1,
+    saying: cat.saying,
+    friends: ['Sam', 'Hanna', 'Kitty', 'Milo']
+}
+
+let messageString = function (item) {
+    let string = `${item.species};${item.name};${item.saying};${item.gender};${item.legs};${item.hands};${item.tail};${item.friends}`;
```
Comment (OleksiyRudenko) [inline submissions/Roophee/a-tyni-js-world-roophee/index.js:57]: Quite a lengthy string.
Imagine you need to add 5 more props at random locations among already existing.
There is also a pattern here (each prop value ends with a semicolon).
Imagine you are asked to add a space after each semicolon.

Can you build a string from a list employing `Array` methods?

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@Roophee now we have a better situation when whatever is the order of properties in an object definition, presentation string will always put values in a particular order as expected.
There is still room to practise `Array` methods to build maintainable and scalable code.

**Review by OleksiyRudenko — APPROVED** [review]
@Roophee great job!

**Pull request merged**